aimd/job/torch_mnist_trainingjob_aimd.py

```diff
@@ -0,0 +1,231 @@
+
+                if args.log_dir and rank == 0:
+                    writer.add_scalar('train/loss', train_loss, global_step)
+
+                if rank == 0:
+                    trial.log(metrics_type='train',
+                              metrics={'loss': train_loss},
+                              step=global_step,
+                              epoch=epoch)
+
+        scheduler.step()
+        global_step = epoch * steps_per_epoch
+        test(val=True, epoch=epoch)
+
+
+def test(val=False, epoch=None):
+    label = 'val' if val else 'test'
+    model.eval()
+    running_loss = 0.0
+    correct = 0
+
+    with torch.no_grad():
+        loader = val_loader if val else test_loader
+        for data, target in loader:
+            data, target = data.to(device), target.to(device)
+            output = model(data)
+            loss = criterion(output, target)
+            running_loss += loss.item()
+            prediction = output.max(1)[1]
+            correct += (prediction == target).sum().item()
+
+    test_loss = running_loss / len(loader)
+    test_accuracy = correct / len(loader.dataset)
+    msg = '{:s} loss: {:.4f}, {:s} accuracy: {:.4f}'.format(
+        label, test_loss, label, test_accuracy)
+    if val:
+        msg = 'epoch {:d}/{:d} with '.format(epoch, epochs) + msg
+    logging.info(msg)
+
+    if args.log_dir and rank == 0:
+        writer.add_scalar('{:s}/loss'.format(label), test_loss, global_step)
+        writer.add_scalar('{:s}/accuracy'.format(label), test_accuracy,
+                          global_step)
+
+    if rank == 0:
+        trial.log(metrics_type=label,
+                  metrics={
+                      'loss': test_loss,
+                      'accuracy': test_accuracy,
+                  },
+                  step=global_step,
+                  epoch=epoch)
+
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+
+    use_cuda = not args.no_cuda and torch.cuda.is_available()
+    if use_cuda:
+        logging.info('Using CUDA')
+    device = torch.device("cuda" if use_cuda else "cpu")
+    kwargs = {'num_workers': 1, 'pin_memory': True} if use_cuda else {}
+
+    logging.info('Using distributed PyTorch with {} backend'.format(
+        args.backend))
+    dist.init_process_group(backend=args.backend)
+    rank = dist.get_rank()
+    world_size = dist.get_world_size()
+
+    params = {
+        'batch_size': 32 * world_size,
+        'epochs': 10,
+        'learning_rate': 0.001 * world_size,
+        'learning_rate_decay_period': 1,
+        'learning_rate_decay_factor': 0.7,
+        'conv_channels1': 32,
+        'conv_channels2': 64,
+        'conv_channels3': 64,
+        'conv_kernel_size': 3,
+        'maxpool_size': 2,
+        'linear_features1': 64,
+        'seed': 1,
+    }
+
+    if rank == 0:
+        from t9k import aimd
+        trial = aimd.create_trial(trial_name='mnist_torch_distributed',
+                                  folder_path='aimd-example')
```

Comment:
Please document in the README that  `folder_path` and `trial_name` are hardcoded in the code ... and where to look for the result in the AIMD UI